sycl/include/sycl/handler.hpp

```diff
@@ -2483,6 +2492,16 @@ class __SYCL_EXPORT handler {
   /// The list of valid SYCL events that need to complete
   /// before barrier command can be executed
   std::vector<detail::EventImplPtr> MEventsWaitWithBarrier;
+  /// Pipe name that uniquely identifies a pipe.
+  std::string HostPipeName;
+  /// Pipe host pointer, the address of its constexpr __pipe member.
+  void *HostPipePtr = nullptr;
+  /// Host pipe read write operation is blocking.
+  bool HostPipeBlocking = false;
+  /// The size of returned type for each read.
+  size_t HostPipeTypeSize = 0;
+  /// If the pipe operation is read or write, 1 for read 0 for write.
+  bool HostPipeRead = true;
```

Comment:
To avoid ABI break in handler (sycl/test/abi/layout_handler.cpp) these should be moved to `handler_impl` instead.

Reply:
Thanks @steffenlarsen. Since this is an experimental API (in sycl\include\sycl\ext\intel\experimental\pipes.hpp), can we get an exception?

Referring to ABI guideline: Features clearly marked as experimental are considered as an exception to this guideline.

Reply:
Sadly we can't make an exception because handler isn't experimental and this affects the ABI even if the user isn't using the extension.

Reply:
Thanks @steffenlarsen for the explanation. Totally understand! I will look into the `handler_impl` to do the move, hopefully that is not complicated.

Reply:
Thanks Steffen. I have moved the code into `handler_impl`.